src/server/session.ts

```diff
@@ -374,7 +374,24 @@ namespace ts.server {
 
         private getCompilerOptionsDiagnostics(args: protocol.CompilerOptionsDiagnosticsRequestArgs) {
             const project = this.getProject(args.projectFileName);
-            return this.convertToDiagnosticsWithLinePosition(project.getLanguageService().getCompilerOptionsDiagnostics(), /*scriptInfo*/ undefined);
+            let diagnostics = this.convertToDiagnosticsWithLinePosition(project.getLanguageService().getCompilerOptionsDiagnostics(), /*scriptInfo*/ undefined);
+
+            if (this.shouldAddTsconfigSuggestionMessage(project)) {
+                const tsconfigDiag = createCompilerDiagnostic(Diagnostics.Adding_a_tsconfig_json_file_will_help_organizing_projects_with_both_TypeScript_and_JavaScript_files_Learn_more_at_https_Colon_Slash_Slashaka_ms_Slashtsconfig);
+                diagnostics = concatenate(diagnostics, this.convertToDiagnosticsWithLinePosition([tsconfigDiag], /*scriptInfo*/ undefined));
+            }
+
+            return diagnostics;
+        }
+
+        private shouldAddTsconfigSuggestionMessage(project: Project) {
+            if (project.projectKind === ProjectKind.External && hasBothJsAndTsFiles(project)) {

```

Comment:
Adding the question at the end of that error message is definitely helpful, this error is almost always the first one what people gets when first switching on `allowJs`. 

In the original issue, the user has to include some JavaScript files in the project because they need to be deployed, however he doesn't want them to be included when compiling TypeScript. As we don't expose `allowJs` in the project property page, currently this can only be achieved by adding a tsconfig file. Therefore we added this message for external project only.

I think these are two different scenarios, but they should all be addressed.